### DISCUSSION TITLE
Ok, I'm lost... I'd like to use, one day even contribute, but can't get rolling...

### DISCUSSION THREAD

**Discussion by elcycliste** [lensfun/lensfun#1799] [General]
OK, I've got darktable and rawtherapee, simply want to test them out for home use on Windows 10 to see which I prefer before making the leap. I can use a computer and am generally technical minded (eng degree), but any mention of code and I'll run away.

I've downloaded the latest lensfun zip release in feb, and have some lenses (e.g. Samyang 75mm) that I believe have only gone into the interim xml release more recently -thankyou to contributors.

I've tried to download the xml, but must be missing something that's obvious to you guys. I've downloaded a "save link as" and also tried going via some 'raw' tab and copy-pasting the xml content in notepad+, with and without the "DOCTYPE..." first line after seeing the file format was different from that in the main-release zip. I then re-opened darktable and it still doesn't recognise the presence of the additional Samyang 75mm lines that have been added to the updated xml file. The updated file starts with version=2 instead of version=1, but still sits in the original "Version 1" darktable/share/lensfun/version 1 directory. Does that matter?

I tried the command line update option I'd read about hoping that'd make life easy if entered in the 'version 1' directory via windows command prompt, but get the impression that only works on linux instead of windows?

And as for rawtherapee, is it correct that it accesses lensfun xml's in the 'darktable' folder as per the rawtherapee instructions? Either which way, rawtherapee doesn't even recognise the lens being on the body, which at least darktable did (just couldn't find the newest profile). but the rawtherapee thing is probably a separate issue unrelated to lensfun.

So in a nutshell, I've spent several hours reading multiple guides, looking for tutorials etc, trying various bits, and if I ever get up and running I'd be happy to provide photos to help setup new profiles for lenses in the future, but without a simple video or equivalent quickly giving a simple walk-though on what is probably a simple update in response to a community effort to get newer lenses in there, I can't get to a happy starting place... <<<How do I do the update please?>>>

Thanks in advance.

**Comment by elcycliste** [comment]
Resolved that rawtherapee issue, turns out it's a bug with Sony camera in release 5.8 and trialling a development release has at least enabled it to read out the Samyang lens, but still can't select the new profile from the drop-down menu.

Main question still stands - how in simple terms do I get add a new xml profile to be seen and acted upon by darktable/rawtherapee etc?

**Comment by rebio** [comment]
@elcycliste 
First you download git, next you pull the source code from the git repo, then you get your coding environment ready!
( Sorry, I just couldn't resist ^^' )

All you should need is the regular windows notepad (even I recommend notepad++ for the reason that it looks nicer and more legible). I only use RawTherapee and I got it working there. Even I want to learn darktable for some time now, I still have no idea here and can't help you with DT. First it didn't work on RT, now it works. I'm not really sure what did change that but I'll have a good guess.

My Steps:
1.  Tried to work on my Linux laptop. Didn't work because of KDE, Dolphin and a secure flatpak folder. (insert meme about how superior Linux is here)
2. Turned over to my Windows PC and went to the path of lensfun in RawTherapee (INSTALL FOLDER\share\lensfun\*)
3. Opened the directory on github (lensfun/data/db/mil-samyang.xml) and displayed it as raw text (https://raw.githubusercontent.com/lensfun/lensfun/master/data/db/mil-samyang.xml)
4. I replaced the content of "mil-samyang.xml" in the RT directory with the one I just copied from github **(I think this step was the mistake because it didn't work and the lens was not displayed)**
5. I copied the content of the **slr**-samyang.xml file in the RT directory to mil-samyang.xml.
6. I deleted every lens data in mil-samyang.xml and only inserted the calibration data for the SAMYANG AF 75mm F1.8 from the raw git file (that starts with `<lens>` and ends with    `</lens>`)
7. I deleted slr-samyang.xml to see if the database is created on startup. If so, only the SAMYANG AF 75mm F1.8 should be visible in the list since there is no data left.
8. Once I restarted RT I only saw the SAMYANG AF 75mm F1.8, could select the lens and apply the calibration data

My guess is that I screwed up the file by copying the raw content from github (I guess I broke some angle bracket or so). Once I just replaced a part of the content of a file that worked, the lens showed up. Of course you don't have to delete the entire content of the calibration file. To move them into a different folder temporary can help you check if it works out.

**EDIT:**
Bug description:
- Bug was caused by Samyang 8mm f/2.8 UMC Fish-eye -> calibration data -> distortion -> "real-focal" value. Once this value was deleted it worked fine
- THIS "real-focal" entry broke every lens beyond this. So RT seemed have caused an error and dropped the file after the first lens

TLDR:
- Download this file: [mil-samyang.txt](https://github.com/lensfun/lensfun/files/8548510/mil-samyang.txt)
-  Rename the .txt file to .xml (github does not allow sharing xml files)
- Replace the current file

**Reply by elcycliste** [reply]
Thank you so much @rebio.
I started with the TLDR file and it worked a treat (tested in Darktable folder - the only difference was when I renamed the file it still kept the txt association, so I replaced the content of the original xml file with the content in your txt file using Notepad++ and that worked out).
Hopefully when I get a bit more time at the weekend I'll try and follow the other steps through to have a better understanding of how to set it up in future.
Thanks again

**Reply by elcycliste** [reply]
Further to this, I've just gone to copy across the updated samyang file to the rawtherapee folder. Turns out the latest development build already has a 75mm profile in, it was just the stock v 5.8 lensfun folder that doesn't.

**Comment by tuxfanx** [comment]
https://github.com/Beep6581/RawTherapee/issues/6463
#1800
